css/background-origin/ tests are not seen by run-webkit-tests

gsnedders opened the pull request:
#### aafee1bd8c1d4abc2794dcaab5edf88b47a4d34c
<pre>
css/background-origin/ tests are not seen by run-webkit-tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=303792">https://bugs.webkit.org/show_bug.cgi?id=303792</a>
<a href="https://rdar.apple.com/166101995">rdar://166101995</a>

Reviewed by Brianna Fan.

This:

 * Fixes TestImporter to remove files from resource-files.json,

 * Removes from resource-files.json everything that appears as a test
   in the WPT manifest,

 * Updates expectations and rebaselines all those tests,

 * Fixes -crash.https.html to be picked up as a crash test, by using
   the WPT manifest logic directly.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/background-clip-padding-box-with-border-radius.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-image-type-001.htm:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-slice-005.htm:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-slice-007.htm:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/history-traversal/unset_context_name-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/utf-16be-expected.txt: Added.
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder.py:
(LayoutTestFinder.is_wpt_crash_test):
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder_legacy_unittest.py:
* Tools/Scripts/webkitpy/port/test.py:
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.__init__):
(TestImporter.find_importable_tests):
(TestImporter.import_tests):
(TestImporter.update_tests_options):
(TestImporter._already_identified_as_resource_file): Deleted.
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:

Canonical link: <a href="https://commits.webkit.org/305533@main">https://commits.webkit.org/305533@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b84a1617d94e89bdf501462fa4af1d15a5c334b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/138687 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/11053 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/169 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/146801 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/91662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/fe052a74-f6bb-4fed-9e89-5dd93715e875) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/140560 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/11757 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/11207 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/106122 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/91662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/696cce8e-0f7c-48dd-b8fa-34ca4431edaf) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/141634 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/8863 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/124301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/86993 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/955f41c3-2c8c-4084-b832-e19470bf1114) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/138034 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/8451 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/6206 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/7099 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/117872 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/141 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/149557 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/10735 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/142 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/114509 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/10752 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/9085 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/114846 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/8696 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/120598 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/65630 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21371 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/10783 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/142 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/10518 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/10721 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/10572 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/Scripts/webkitpy/w3c/test_importer.py:
```diff
@@ -431,6 +432,8 @@ def find_importable_tests(self, directory):
                 elif self._is_in_resources_directory(fullpath):
                     _log.warning('%s is a test located in a "resources" folder. This test will be skipped by WebKit test runners.', fullpath)
 
+                self._non_test_resource_files.append(fullpath)
```
kind of nit: Does this read as "non-test resource file" or "non test-resource files"? Can we rename this since it's a bit confusing to the reader? Personally I keep thinking "a resource file that is not a test type" rather than "not a resource file".

briannafan commented on Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
```diff
@@ -332,6 +332,29 @@ def test_update_slow_test(self):
         tests_options = json.loads(fs.read_text_file('/mock-checkout/LayoutTests/tests-options.json'))
         self.assertIn("slow", tests_options["imported/w3c/web-platform-tests/a/existing-test.html"])
 
+    def test_remove_now_test_resource_file(self):
```
Should `now` be `non`?

briannafan reviewed (commented):
It looks like the test failures aren't related to your PR, but could you double check for the WPT tests in particular?

Also, I assume we always ignore the style check for `port/test.py`?

gsnedders committed:
css/background-origin/ tests are not seen by run-webkit-tests
https://bugs.webkit.org/show_bug.cgi?id=303792
rdar://166101995

Reviewed by Brianna Fan.

This:

 * Fixes TestImporter to remove files from resource-files.json,

 * Removes from resource-files.json everything that appears as a test
   in the WPT manifest,

 * Updates expectations and rebaselines all those tests,

 * Fixes -crash.https.html to be picked up as a crash test, by using
   the WPT manifest logic directly.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/background-clip-padding-box-with-border-radius.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-image-type-001.htm:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-slice-005.htm:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/border-image-slice-007.htm:
* LayoutTests/imported/w3c/web-platform-tests/html/browsers/browsing-the-web/history-traversal/unset_context_name-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/utf-16be-expected.txt: Added.
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder.py:
(LayoutTestFinder.is_wpt_crash_test):
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder_legacy_unittest.py:
* Tools/Scripts/webkitpy/port/test.py:
* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.__init__):
(TestImporter.find_importable_tests):
(TestImporter.import_tests):
(TestImporter.update_tests_options):
(TestImporter._already_identified_as_resource_file): Deleted.
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:

Canonical link: https://commits.webkit.org/305533@main
